### ISSUE TITLE
False positive when assigning to a property with a setter that takes `T | None`

### ISSUE THREAD

**Issue by StockerMC** [microsoft/pyright#3354]
**Describe the bug**
When assigning `Any | None` to a property setter that takes `T | None` (`T` being any type), pyright fails with an error as shown below.

**Screenshots or Code**
```python
import datetime

class Embed:
    @property
    def timestamp(self) -> datetime.datetime | None:
        ...

    @timestamp.setter
    def timestamp(self, value: datetime.datetime | None) -> None:
        ...

def set_timestamp(embed: Embed, data: dict[str, Any]):
    timestamp = data.get('timestamp')
    reveal_type(timestamp)  # Any | None
    # Cannot assign member "timestamp" for type "Embed"
    # "Embed | None" is incompatible with "datetime | None"
    embed.timestamp = timestamp
```

**VS Code extension or command-line**
This affects pyright 1.1.237 and Pylance v2022.4.1.

**Additional context**
```python
class Embed:
    @property
    def timestamp(self) -> int:
        ...

    @timestamp.setter
    def timestamp(self, value: int) -> None:
        ...

def set_timestamp(embed: Embed, data: dict[str, Any]):
    timestamp = data.get('timestamp')
    reveal_type(timestamp)  # Any | None
    # Cannot assign member "timestamp" for type "Embed"
    # "Embed | None" is incompatible with "int"
    embed.timestamp = timestamp
```
Pyright is correct to emit an error here, but the error appears to be incorrect.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is fixed in pyright 1.1.238, which I just published. It will also be included in the next release of pylance.
